src/Fetcher/TululaFetcher.php

```diff
@@ -0,0 +1,241 @@
+use Psr\Log\LoggerInterface;
+use Symfony\Component\String\Slugger\SluggerInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class TululaFetcher implements FetcherInterface
+{
+    const SOURCE = 'tulula';
+    private const TAGS_SYNONYMS = [
+        'android',
+        'css',
+        'data',
+        'devops',
+        'dotnet',
+        'facebook',
+        'golang',
+        'html',
+        'ios',
+        'java',
+        'javascript',
+        'nodejs',
+        'php',
+        'python',
+        'react native',
+        'ruby',
+        'rust',
+        'security',
+        'ux',
+        'ai/ml',
+        'sql',
+        'iot',
+    ];
+
+    private LoggerInterface $logger;
+    private TagRepository $tagRepository;
+    /** @var ExcludedTag[] */
+    private array $excludedTags;
+    private LocationGuesser $locationGuesser;
+    private SluggerInterface $slugger;
+    private HttpClientInterface $client;
+
+    public function __construct(LoggerInterface $logger, LocationGuesser $locationGuesser, HttpClientInterface $client, TagRepository $tagRepository, ExcludedTagRepository $excludedTagRepository, SluggerInterface $slugger)
+    {
+        $this->client = $client;
+        $this->logger = $logger;
+        $this->tagRepository = $tagRepository;
+        $this->locationGuesser = $locationGuesser;
+        $this->slugger = $slugger;
+        $this->excludedTags = $excludedTagRepository->findAll();
+    }
+
+    public function isActive(): bool
+    {
+        return true;
+    }
+
+    public function getUrl(array $params = []): string
+    {
+        return 'https://tulu.la/api/public';
+    }
+
+    public function fetch(): \Generator
+    {
+        foreach ($this->tagRepository->findBy(['selected' => true]) as $tag) {
+            $tagKey = array_search($this->slugger->slug(strtolower($tag->getName()))->toString(), self::TAGS_SYNONYMS);
+
+            if (false === $tagKey) {
+                continue;
+            }
+
+            $tagSynonym = self::TAGS_SYNONYMS[$tagKey];
+            $sameTagConferences = [];
+            foreach ($this->queryTululaEvents() as $conference) {
+                foreach ($conference['tags'] as $conferenceTag) {
+                    if ($tagSynonym === strtolower($conferenceTag['name'])) {
+                        $sameTagConferences[] = $conference;
+                        break;
+                    }
+                }
+            }
+
+            yield $this->getUrl() => $this->denormalizeConferences($sameTagConferences, $tag);
+        }
+    }
+
+    public function denormalizeConferences(array $rawConferences, Tag $tag): array
+    {
+        $conferences = [];
+        foreach ($rawConferences as $rawConference) {
+            $city = null;
+            if (!$rawConference['isOnline']) {
+                if ($city = $rawConference['venue']['city'] ?: $rawConference['venue']['state']) {
```

Comment:
Is it possible to have an falsy value here?

Reply:
It is possible to have an empty string, which will result in the city being null.
Do you think we should add a message or handle this case better ?